code/ModelAdmin.php

```diff
@@ -150,22 +150,27 @@ protected function init()
         $models = $this->getManagedModels();
         $this->modelTab = $this->getRequest()->param('ModelClass');
 
+        // security check for valid models
+        if ($this->modelTab && !$this->isManagedModel($this->modelTab)) {
+            // No need to throw exceptions since we are already redirected
```

Comment:
```suggestion
            // No need to check model tab since we are already redirected
            // This happens when there's a permission failure in LeftAndMain
```
Just a small change to make it really clear why this is here - otherwise I could see this being removed if this code is refactored in the future.